components/webext-storage/src/api.rs

```diff
@@ -62,6 +65,33 @@ fn remove_from_db(tx: &Transaction<'_>, ext_id: &str) -> Result<()> {
     Ok(())
 }
 
+/// A newtype for a stored value, which dereferences to its contents. This
```

Comment:
I gotta say I'm not completely loving this new type. Magic `IntoFfi` is nice and all, but the changes below with needing to do the magic `&*` dances adds a layer of complexity and opaqueness that doesn't really seem like a net win.

But needing this wrapper type so `implement_into_ffi_by_json` can be used on a JsonValue doesn't sound right, so I suspect I'm missing something?

Reply:
I agree—I'm tempted to revert my changes to `StorageChanges` for the same reason, it feels too magical compared to what you had before. Let's wait for @thomcc to chime in, I probably missed a much easier way to do what we had before!

Reply:
(I totally did—just call `serde_json::to_string` directly in the FFI code! 🤦‍♀️)